client/src/connection/ssh/index.ts

```diff
@@ -25,175 +18,199 @@ export interface Config {
+                  this.clearTimer();
+                  this.resolve?.();
+                }
               }
             });
-          } else {
-            logs.push(output);
-            if (output.endsWith("?")) {
-              resolve?.();
-            } else {
-              //we're not in running state, set a timeout so we dont attempt to connect indefinitely
-              setTimeout(() => {
-                reject?.(
-                  new Error(
-                    "Execution Failed. Verify that Profile connection settings are correct."
-                  )
-                );
-                logs = [];
-                conn.end();
-              }, sasLaunchTimeout);
-            }
-          }
-        });
 
-      const resolvedEnv: string[] = [
-        '_JAVA_OPTIONS="-Djava.awt.headless=true"',
-      ];
-      const execArgs: string = resolvedEnv.join(" ");
+            const resolvedEnv: string[] = [
+              '_JAVA_OPTIONS="-Djava.awt.headless=true"',
+            ];
+            const execArgs: string = resolvedEnv.join(" ");
 
-      const resolvedSasOpts: string[] = [
-        "-nodms",
-        "-terminal",
-        "-nosyntaxcheck",
-      ];
+            const resolvedSasOpts: string[] = [
+              "-nodms",
+              "-terminal",
+              "-nosyntaxcheck",
+            ];
 
-      if (config.sasOptions) {
-        resolvedSasOpts.push(...config.sasOptions);
-      }
-      const execSasOpts: string = resolvedSasOpts.join(" ");
+            if (this.config.sasOptions) {
+              resolvedSasOpts.push(...this.config.sasOptions);
+            }
+            const execSasOpts: string = resolvedSasOpts.join(" ");
 
-      stream.write(`${execArgs} ${config.saspath} ${execSasOpts} \n`);
-    });
-  })
-  .on("error", (err) => {
-    reject?.(err);
-    return;
-  });
-
-function getResult() {
-  const runResult: RunResult = {};
-  if (!html5FileName) {
-    resolve?.(runResult);
-    return;
-  }
-  let fileContents = "";
-  conn.exec(`cat ${html5FileName}.htm`, (err: Error, s: ClientChannel) => {
-    if (err) {
-      reject?.(err);
-      return;
-    }
+            this.stream.write(
+              `${execArgs} ${this.config.saspath} ${execSasOpts} \n`
+            );
+          });
+        })
+        .on("error", (err) => {
+          this.clearTimer();
+          this.reject?.(err);
+          return;
+        });
 
-    s.on("data", (data) => {
-      fileContents += data.toString().trimEnd();
-    }).on("close", (code) => {
-      const rc: number = code;
-
-      if (rc === 0) {
-        //Make sure that the html has a valid body
-        //TODO: should this be refactored into a shared location?
-        if (fileContents.search('<*id="IDX*.+">') !== -1) {
-          runResult.html5 = fileContents;
-          runResult.title = "Result";
-        }
-      }
-      resolve?.(runResult);
+      this.setTimer();
+      this.conn.connect(cfg);
     });
-  });
-}
+  };
```

Comment:
Can we add new lines between methods? Also, I noticed you used a mix of "traditional" (`a() {}`) style functions and arrow function (`a: () => {}`). Would you mind making all methods here consistent?

Reply:
The arrow function was necessary when dealing with the promise return so that the "this" context is properly defined so that I can set state on the instance. Will update the rest to use arrow functions to be consistent.